src/components/Modal/Modal.tsx

```diff
@@ -74,8 +74,13 @@ export const Modal = ({
     }
   };
 
+  const handleEscKey = (e: KeyboardEvent<HTMLDivElement>) => {
+    e.stopPropagation();
```

Comment:
I think using `stopImmediatePropagation()` may be safer here.

Other key events may be handled on the same element (document), so I think using `stopImmediatePropagation()` would prevent any other events be triggered.

Reply:
Good point! I have changed that @ [94c5a54](https://github.com/canonical/react-components/commit/94c5a54914d1d1a429b04240dd03aefaba6a5eb2)